pynamodb/models.py

```diff
@@ -551,7 +551,10 @@ def from_raw_data(cls: Type[_T], data: Dict[str, Any]) -> _T:
             attr_name = cls._dynamo_to_python_attr(name)
             attr = cls.get_attributes().get(attr_name, None)  # type: ignore
             if attr:
-                attributes[attr_name] = attr.deserialize(attr.get_value(value))  # type: ignore
+                try:
+                    attributes[attr_name] = attr.deserialize(attr.get_value(value))  # type: ignore
+                except TypeError:
+                    raise TypeError(f'attribute: {attr_name}')
```

Comment:
I would consider introducing a new pynamo exception here that we can attach the attribute name context to, and do `raise SomePynamoExc() from e` to maintain the original `TypeError` context.

@ikonst thoughts?

Reply:
Agree, let's add it :)

Reply:
@jonathantan let me know if this is clear, otherwise I can give more details!

Reply:
Interesting I'm learning exception chaining for the first time but it sounds pretty straight forward according to https://www.python.org/dev/peps/pep-3134/#explicit-exception-chaining

Reply:
done :)